### PR TITLE
Adds Tests for WriterAsOutputStream

### PR DESCRIPTION
**Pull request by LeonardKrml**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)
#489 

### What is the new behavior?
added tests

### Does this PR introduce a breaking change? (check one with "x")
- [ ] Yes
- [x] No

###  Other information

### CODE REVIEW

**Review by icaOBU — APPROVED** [review]

File: tests/Yaapii.Atoms.Tests/IO/WriterAsOutputStreamTest.cs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    {
+        [Fact]
+        public void WritesContentToFile()
+        {
+            var dir = "artifacts/WriterAsOutputStreamTest"; var inputFile = "large-text.txt";
+            var inputPath = Path.GetFullPath(Path.Combine(dir, inputFile));
+            var outputFile = "text-copy.txt";
+            var outputPath = Path.GetFullPath(Path.Combine(dir, outputFile));
+
+            Directory.CreateDirectory(dir);
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/IO/WriterAsOutputStreamTest.cs:43]: This leaves the directory present after the test. We always want to cleanup after a test (and be sure that we do not test data from a previous test). Please have a look at the object TempDirectory and TempFile and use them here.

File: tests/Yaapii.Atoms.Tests/IO/WriterAsOutputStreamTest.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            ).Value();
+
+            long right =
+                new LengthOf(
+                    new InputOf(
+                        new Uri(Path.GetFullPath(outputPath))
+                    )
+                ).Value();
+
+            Assert.True(left == right, "input and output are not the same size");
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/IO/WriterAsOutputStreamTest.cs:93]: Put the objects left and right directly into the Assert:
`Assert.True(new LengthOf(...).Value() == new LenghtOf(...).Value(), ...)`

File: tests/Yaapii.Atoms.Tests/IO/WriterAsOutputStreamTest.cs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        }
+
+        [Fact]
+        public void TryRead()
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/IO/WriterAsOutputStreamTest.cs:97]: Variable name must be a fact about the tested object. This is a fact what the test method does. Please rename to "RejectsReading".

File: tests/Yaapii.Atoms.Tests/IO/WriterAsOutputStreamTest.cs [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+                () => stream.Read(bytes,0,1));
+        }
+
+        [Fact]
+        public void TrySeek()
+        {
+            var stream = new WriterAsOutputStream(
+                             new StreamWriter(new MemoryStream())
+                             );
+            Assert.ThrowsAny<NotImplementedException>(
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/IO/WriterAsOutputStreamTest.cs:113]: Rename to RejectsSeeking

File: tests/Yaapii.Atoms.Tests/IO/WriterAsOutputStreamTest.cs [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+                () => stream.Seek(3L, SeekOrigin.Begin));
+        }
+
+        [Fact]
+        public void TrySetLength()
+        {
+            var stream = new WriterAsOutputStream(
```
Comment (Meerownymous) [inline tests/Yaapii.Atoms.Tests/IO/WriterAsOutputStreamTest.cs:120]: Do not declare a variable.

**Review by Meerownymous — CHANGES_REQUESTED** [review]

**Review by Meerownymous — APPROVED** [review]

**Pull request merged**